### PR TITLE
FOIA-239: Document FOIA export xml.

### PR DESCRIPTION
**Pull request by jesseday**

### CODE REVIEW

File: docroot/modules/custom/foia_export_xml/README.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+can be found documented on [github](https://github.com/usdoj/foia-api/blob/develop/docs/FoiaAnnualReportExtensions.xsd).
+
+#### Exporting Annual FOIA Report Data nodes
+
+The module defines a route with the pattern `/node/[nid]/xml` and a menu link
+that displays on node pages.  The route's controller builds the XML from node
+data and returns a response with the export contents as a file attachment for
+download.
+
+#### The export class
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Some h4s are using sentence casing and others are using title casing. Can all be updated to use sentence casing?

File: docroot/modules/custom/foia_export_xml/README.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+#### The export class 
+
+The bulk of the work in this module is done in the class 
+`\Drupal\foia_export_xml\ExportXml`.  This class builds an XML document from
+ node data that conforms to the FOIA Annual Report schema extension.  The
+`__toString()` method is used to convert the XML document to a string.  In
+this method, a new export can be created, then cast to a string as the body
+of the response object.
+
+#### Export Structure
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Change to sentence casing.

File: docroot/modules/custom/foia_export_xml/README.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  multiple types of data.
+* `ReliedUponStatuteOrganizationAssociation`: An organization association
+which maps component data to a specific organization within the reporting
+agency.
+
+\* This is not a complete example.  For exact specifications, reference the
+[FOIA Annual Report](https://github.com/usdoj/foia-api/blob/develop/docs/FoiaAnnualReportExtensions.xsd)
+schema.
+  
+#### Agency Components
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Change to sentence casing. Even though both are capitalized when referring to the node type, I think title-wise we're addressing to in general so sentence casing is good.

File: docroot/modules/custom/foia_export_xml/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    <nc:OrganizationName>United States Department of Agriculture</nc:OrganizationName>
+    <nc:OrganizationSubUnit s:id="ORG1">
+        <nc:OrganizationAbbreviationText>AMS</nc:OrganizationAbbreviationText>
+        <nc:OrganizationName>Agricultural Marketing Service</nc:OrganizationName>
+    </nc:OrganizationSubUnit>
+    ...
+</nc:Organization>
+```  
+
+##### Component Data
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Change to sentence casing.

File: docroot/modules/custom/foia_export_xml/README.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+The `componentMap` can then be used to retrieve an organizational id during
+export based on a referenced Agency Component's node id. For example:
+
+```
+$agency_component = $component->field_agency_component->referencedEntities()[0];
+$identifier = $this->componentMap[$agency_component->id()]
+```
+
+#### Centralized vs Non-Centralized Agencies
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Change to sentence casing.

File: docroot/modules/custom/foia_export_xml/README.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+[FOIA Annual Report](https://github.com/usdoj/foia-api/blob/develop/docs/FoiaAnnualReportExtensions.xsd)
+schema.
+  
+#### Agency Components
+
+Agencies can consist of one or more subunits, referred to as components. These
+exist in Drupal as Agency Component nodes.  When creating an Annual FOIA
+Report Data node, paragraph items containing component data will reference
+one of the Agency's components, associating the data in that paragraph item
+with a specific organizational subunit.
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: I know it's noted in the centralized vs non-centralized agency section, but might be worth noting here as well that an agency must consist of at least one component (the component being the agency itself).

File: docroot/modules/custom/foia_export_xml/README.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
+INTRODUCTION
+------------
+
+The main task of the FOIA Export XML module is to transform an Annual FOIA
+Report Data node into an XML file that conforms to the FOIA Annual Report
+extension of the NIEM IEPD schema.  A representation of this extension schema
+can be found documented on [github](https://github.com/usdoj/foia-api/blob/develop/docs/FoiaAnnualReportExtensions.xsd).
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Can we change this to say "GitHub"?

File: docroot/modules/custom/foia_export_xml/README.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+This module depends on the `annual_foia_report_data` and `agency_component`
+content types.
+
+INSTALLATION
+------------
+
+Install as you would normally install a contributed Drupal module. Visit
+https://www.drupal.org/docs/8/extending-drupal-8/installing-drupal-8-modules#s-step-2-enable-the-module
+for further information.
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Just change this wording to be consistent with Aaron's:

FOIA Export XML is a custom Drupal module so unlike contrib modules, the codebase is not
installed via composer. Enable as you would normally enable a contributed Drupal
module. Visit
https://www.drupal.org/docs/8/extending-drupal-8/installing-drupal-8-modules#s-step-2-enable-the-module
for futher information.

File: docroot/modules/custom/foia_export_xml/README.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+CONFIGURATION
+-------------
+
+Any user with permission to view an Annual FOIA Report Data node, also has
+permission to export that node to XML.  Access depends on whether or not the 
+node is published.
+
+Configure the user permissions in Administration » People » Permissions:
+
+* View published content
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: I can't tell here, but for bullets there should be a space before the asterisk.

[space][asterisk][space][text]

File: docroot/modules/custom/foia_export_xml/README.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+Install as you would normally install a contributed Drupal module. Visit
+https://www.drupal.org/docs/8/extending-drupal-8/installing-drupal-8-modules#s-step-2-enable-the-module
+for further information.
+
+CONFIGURATION
+-------------
+
+Any user with permission to view an Annual FOIA Report Data node, also has
+permission to export that node to XML.  Access depends on whether or not the 
+node is published.
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: > Access depends on whether or not the node is published.

I don't believe the node has to be published in order to export the XML. We can probably just take that line out.

File: docroot/modules/custom/foia_export_xml/README.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+```
+<foia:Exemption3StatuteSection>
+    <foia:ReliedUponStatute s:id="ES2">...</foia:ReliedUponStatute>
+    ...
+    <foia:ReliedUponStatuteOrganizationAssociation>...</foia:ReliedUponStatuteOrganizationAssociation>
+</foia:Exemption3StatuteSection>
+```  
+
+In this example, the elements are*:
+* `Exemption3StatuteSection`: A single section of the report.
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: I can't tell here, but for bullets there should be a space before the asterisk.

[space][asterisk][space][text]

File: docroot/modules/custom/foia_export_xml/README.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+agency itself. For such agencies, agency-overall data is not exported to XML
+since report since that data will be contained within component data.
+
+
+REQUIREMENTS
+------------
+
+This module depends on the `annual_foia_report_data` and `agency_component`
+content types.
+
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Need two empty lines prior to headings (except first one).

File: docroot/modules/custom/foia_export_xml/README.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
+INTRODUCTION
+------------
+
+The main task of the FOIA Export XML module is to transform an Annual FOIA
+Report Data node into an XML file that conforms to the FOIA Annual Report
+extension of the NIEM IEPD schema.  A representation of this extension schema
+can be found documented on [github](https://github.com/usdoj/foia-api/blob/develop/docs/FoiaAnnualReportExtensions.xsd).
+
+#### Exporting Annual FOIA Report Data nodes
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: "Introduction" looks to be an h2, so let's change this to an h3 instead of an h4

Comment (bemarlan) [reply docroot/modules/custom/foia_export_xml/README.md]: Same with other headings.

File: docroot/modules/custom/foia_export_xml/README.md [hunk 14]
```diff
@@ -0,0 +1,164 @@
+INTRODUCTION
```
Comment (bemarlan) [inline docroot/modules/custom/foia_export_xml/README.md]: Can you add ToC up here?

File: docroot/modules/custom/foia_export_xml/README.md (hunk 10 above)
Comment (jesseday) [reply docroot/modules/custom/foia_export_xml/README.md]: I meant it in the sense that some users have access to published nodes while others don't, but I see how that sounds confusing.  I'll change it.

**Pull request merged**